api_tests/nodes/views/test_node_relationship_institution.py

```diff
@@ -0,0 +1,159 @@
+from nose.tools import *
+
+from framework.auth.core import Auth
+
+from tests.base import ApiTestCase
+from tests.factories import InstitutionFactory, AuthUserFactory, NodeFactory
+
+from api.base.settings.defaults import API_BASE
+
+class TestNodeRelationshipInstitution(ApiTestCase):
+    def setUp(self):
+        super(TestNodeRelationshipInstitution, self).setUp()
+        self.user = AuthUserFactory()
+        self.institution = InstitutionFactory()
+        self.node = NodeFactory()
+        self.node_institution_url = '/{0}nodes/{1}/relationships/institution/'.format(API_BASE, self.node._id)
+
+    def test_node_with_no_permissions(self):
+        self.user.affiliated_institutions.append(self.institution)
+        self.user.save()
+        res = self.app.put_json_api(
+            self.node_institution_url,
+            {'data': {'type': 'institution', 'id': self.institution._id}},
+            expect_errors=True,
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 403)
+
+    def test_user_with_no_institution(self):
+        node = NodeFactory(creator=self.user)
+        res = self.app.put_json_api(
+            '/{0}nodes/{1}/relationships/institution/'.format(API_BASE, node._id),
+            {'data': {'type': 'institution', 'id': self.institution._id}},
+            expect_errors=True,
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 403)
+
+    def test_institution_doesnt_exist(self):
+        node = NodeFactory(creator=self.user)
+        res = self.app.put_json_api(
+            '/{0}nodes/{1}/relationships/institution/'.format(API_BASE, node._id),
+            {'data': {'type': 'institution', 'id': 'not_an_id'}},
+            expect_errors=True,
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 404)
+
+    def test_wrong_type(self):
+        self.user.affiliated_institutions.append(self.institution)
+        self.user.save()
+        node = NodeFactory(creator=self.user)
+        res = self.app.put_json_api(
+            '/{0}nodes/{1}/relationships/institution/'.format(API_BASE, node._id),
+            {'data': {'type': 'not_institution', 'id': self.institution._id}},
+            expect_errors=True,
+            auth=self.user.auth
+        )
+
+        assert_equal(res.status_code, 500)
```

Comment:
This should be a 409 Conflict.  Putting `type=TypeField(...)` on the NodeInstitutionRelationshipSerializer will take care of this.